examples/user_interfaces/toolmanager.py

```diff
@@ -83,10 +114,21 @@ def set_lines_visibility(self, state):
 fig.canvas.manager.toolbar.add_tool('zoom', 'foo')
 
 # Remove the forward button
-fig.canvas.manager.toolmanager.remove_tool('forward')
+tool_mgr.remove_tool('forward')
 
 # To add a custom tool to the toolbar at specific location inside
 # the navigation group
 fig.canvas.manager.toolbar.add_tool('Hide', 'navigation', 1)
 
+for i, c in enumerate(['yellowgreen', 'forestgreen']):
+    sidebar = fig.canvas.manager._get_toolbar()
```

Comment:
For this example it works, because we 100% know which backend we use but I want to create it more generic than that, even for my own testing purposes I change the ``matplotlib.use('GTK3Cairo')`` line to quickly check the example works on a different backend, as I have gone through refactoring the various backends my testing file now has grown to have use lines for each backend which them all commented out apart from the backend I want to test.  For other people I want to promote generic code so that they don't curse when they want to change their default backend for some reason (perhaps a nice new shiny backend comes out, I don't want them to change all their code) to adapt.

The other reason, comes from needed access to the ToolManager used for the figure so the minimal amount of code to do what you suggest looks like:
```python
from matplotlib.backends.backend_gtk3 import ToolbarGTK3
ToolbarGTK3(fig.canvas.manager.toolmanager)
```

Alternatively we have
```python
fig.canvas.manager.get_backend().Toolbar(fig.canvas.manager.toolmanager)
```

or as I do above with a method rename...
```python
fig.canvas.manager.new_toolbar()
```
If we go along the route of the latter perhaps we should have the manager keep a list of toolbars?  That would make sense for the ``FigureManager.destroy`` method.